photon-lib/src/main/java/org/photonvision/PhotonUtils.java

```diff
@@ -206,4 +213,50 @@ public static Rotation2d getYawToPose(Pose2d robotPose, Pose2d targetPose) {
     public static double getDistanceToPose(Pose2d robotPose, Pose2d targetPose) {
         return robotPose.getTranslation().getDistance(targetPose.getTranslation());
     }
+
+    /**
+     * Corrects a given pixel for perspective distortion
+     *
+     * @param pixel pixel to be corrected
+     * @param camIntrinsics the distortion matrix of the camera
+     * @return the angles
+     */
+    public static Rotation3d correctPixelRot(Point pixel, Matrix<N3, N3> camIntrinsics) {
+        double fx = camIntrinsics.get(0, 0);
+        double cx = camIntrinsics.get(0, 2);
+        double xOffset = cx - pixel.x;
+
+        double fy = camIntrinsics.get(1, 1);
+        double cy = camIntrinsics.get(1, 2);
+        double yOffset = cy - pixel.y;
+
+        // calculate yaw normally
+        var yaw = new Rotation2d(fx, xOffset);
+        // correct pitch based on yaw
+        var pitch = new Rotation2d(fy / Math.cos(Math.atan(xOffset / fx)), -yOffset);
+
+        return new Rotation3d(0, pitch.getRadians(), yaw.getRadians());
+    }
+
+
+
+    private static final TimeInterpolatableBuffer<Rotation2d> headingBuffer = TimeInterpolatableBuffer.createBuffer(
+            1.0
+    );
```

Comment:
Photonutils should be stateless.

Reply:
Should I create a separate class to hold the heading buffer to keep PhotonUtils stateless, or would it be best to move that logic into a static function and variable PhotonPoseEstimator?